File: deployment/src/main/java/io/quarkiverse/jimmer/deployment/graphql/codegen/JimmerGraphQLSourceType.java
package io.quarkiverse.jimmer.deployment.graphql.codegen;

import java.util.List;

record JimmerGraphQLSourceType(
        String packageName,
        String simpleName,
        String qualifiedName,
        JimmerGraphQLSourceKind kind,
        List<String> imports,
        List<String> extendsTypes,
        List<JimmerGraphQLSourceMethod> methods) {

    boolean isEntity() {
        return kind == JimmerGraphQLSourceKind.ENTITY;
    }

    boolean isMappedSuperclass() {
        return kind == JimmerGraphQLSourceKind.MAPPED_SUPERCLASS;
    }
}

File: ksp-processor/src/main/java/io/quarkiverse/jimmer/graphql/ksp/JimmerGraphQLSourceType.kt
package io.quarkiverse.jimmer.graphql.ksp

data class JimmerGraphQLSourceType(
    val packageName: String,
    val simpleName: String,
    val qualifiedName: String,
    val kind: JimmerGraphQLSourceKind,
    val extendsTypes: List<String>,
    val methods: List<JimmerGraphQLSourceMethod>,
) {

    fun isEntity(): Boolean = kind == JimmerGraphQLSourceKind.ENTITY

    fun isMappedSuperclass(): Boolean = kind == JimmerGraphQLSourceKind.MAPPED_SUPERCLASS
}
